plugins/parsers/json_v2/parser.go

```diff
@@ -335,22 +381,56 @@ func (p *Parser) expandArray(result MetricNode) ([]telegraf.Metric, error) {
 	return results, nil
 }
 
+func (p *Parser) existsInpathResults(index int, raw string) *PathResult {
+	for _, f := range p.pathResults {
+		if f.result.Index == 0 {
+			for _, i := range f.result.Indexes {
+				if i == index {
+					return &f
+				}
+			}
+		} else if f.result.Index == index {
+			return &f
+		}
+	}
+	return nil
+}
+
 // processObjects will iterate over all 'object' configs and create metrics for each
-func (p *Parser) processObjects(objects []JSONObject, input []byte) ([]telegraf.Metric, error) {
+func (p *Parser) processObjects(objects []JSONObject) ([]telegraf.Metric, error) {
 	p.iterateObjects = true
 	var t []telegraf.Metric
 	for _, c := range objects {
 		p.currentSettings = c
+
 		if c.Path == "" {
 			return nil, fmt.Errorf("GJSON path is required")
 		}
-		result := gjson.GetBytes(input, c.Path)
+		result := gjson.GetBytes(p.InputJSON, c.Path)
+
+		// hastag doesn't return index! idea: replace all hastags with index, and find lenght of array
```

Comment:
```suggestion
		// hashtag doesn't return index! idea: replace all hashtags with index, and find length of array
```

I'm not sure if this comment is supposed to say hashtag or if it's still needed

Reply:
Your right this shouldn't be here. Removed.